### PR TITLE
964 dbsql warehouse docs

### LINKED ISSUE #964: [DOCS] - Add DBSQL Warehouse
Add documents for DBSQL Warehouse

### PR DESCRIPTION
**Pull request by aman-db**
closes #964

### CODE REVIEW

**Review by sriram251-code — APPROVED** [review]

File: docs-compose/overwatch_docs/content/DataEngineer/Pipeline_Management.md [hunk 1]
```diff
@@ -133,52 +133,55 @@ when they occur. Sometimes, there will truly be no new data in an upstream Modul
...
+| 3010      | Gold_SparkJob                       | sparkjob_gold                      | 2006                     | 0.6.0   |
+| 3011      | Gold_SparkStage                     | sparkstage_gold                    | 2007                     | 0.6.0   |
+| 3012      | Gold_SparkTask                      | sparktask_gold                     | 2008                     | 0.6.0   |
+| 3013      | Gold_SparkExecution                 | sparkexecution_gold                | 2005                     | 0.6.0   |
+| 3014      | Gold_SparkExecutor                  | sparkexecutor_gold                 | 2003                     | 0.6.0   |
+| 3016      | Gold_SparkStream                    | sparkstream_gold                   | 1006,2005                | 0.7.0   |
+| 3015      | Gold_jobRunCostPotentialFact        | jobruncostpotentialfact_gold       | 3001,3003,3005,3010,3012 | 0.6.0   |
+| 3017      | Gold_Sql_QueryHistory               | sql_query_history_gold             | 2020                     | 0.7.0   |
+| 3018      | Gold_Warehouse                      | warehouse_gold                     | 2021                     | 0.7.2   |
```
Comment (gueniai) [inline docs-compose/overwatch_docs/content/DataEngineer/Pipeline_Management.md:184]: version should be 0.7.2.1 for all the tables we're adding now

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md [hunk 2]
```diff
@@ -0,0 +1,1005 @@
...
+| warehouse_name            | string  | User-defined name of the warehouse                                                                                                                                                                                        |                                                                                                                    
+| service_name              | string  | Name of the service corresponding to DBSQL warehouse                                                                                                                                                                      |                                                                                                                   
+| action_name               | string  | create, edit, or snapImpute – depicts the type of action for the cluster – **snapImpute is used on first run to initialize the state of the cluster even if it wasn’t created/edited since audit logs began               |
```
Comment (gueniai) [inline docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:908]: replace cluster with warehouse

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md [hunk 3]
```diff
@@ -0,0 +1,1005 @@
...
+| action_name               | string  | create, edit, or snapImpute – depicts the type of action for the cluster – **snapImpute is used on first run to initialize the state of the cluster even if it wasn’t created/edited since audit logs began               |              
+| user_email                | string  | Email of the user requesting the action                                                                                                                                                                                   |                                                                                                                                                                                  
+| cluster_size              | string  | Size of the clusters allocated for this warehouse. Increasing the size of a spark cluster allows you to run larger queries on it. If you want to increase the number of concurrent queries, please tune max_num_clusters  |
```
Comment (gueniai) [inline docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:910]: I don't think we should give perf. tuning advice here

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md [hunk 4]
```diff
@@ -0,0 +1,1005 @@
...
+| cluster_size              | string  | Size of the clusters allocated for this warehouse. Increasing the size of a spark cluster allows you to run larger queries on it. If you want to increase the number of concurrent queries, please tune max_num_clusters  | 
+| min_num_clusters          | long    | Minimum number of available clusters that will be maintained for this SQL warehouse.                                                                                                                                      | 
+| max_num_clusters          | long    | Maximum number of clusters that the autoscaler will create to handle concurrent querie                                                                                                                                    |
```
Comment (gueniai) [inline docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:912]: missing s at the end

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md [hunk 5]
```diff
@@ -0,0 +1,1005 @@
...
+| auto_stop_mins            | long    | The amount of time in minutes that a SQL warehouse must be idle (i.e., no RUNNING queries) before it is automatically stopped.                                                                                            | 
+| spot_instance_policy      | string  | Configurations whether the warehouse should use spot instances.                                                                                                                                                           | 
+| enable_photon             | boolean | Configures whether the warehouse should use Photon optimized clusters.                                                                                                                                                    | 
+| channel                   | struct  | This column contains channel details. Some examples - CHANNEL_NAME_UNSPECIFIED, CHANNEL_NAME_PREVIEW, CHANNEL_NAME_CURRENT                                                                                                |
```
Comment (gueniai) [inline docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:916]: If there are more options here, where can they read more about it? Can we link to docs?

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md [hunk 6]
```diff
@@ -0,0 +1,1005 @@
...
+| enable_serverless_compute | boolean | Flag indicating whether the warehouse should use serverless compute.                                                                                                                                                      |                                                                                                                                                     
+| warehouse_type            | string  | Warehouse type: PRO or CLASSIC                                                                                                                                                                                            |                                                                                                                                                                                           
+| warehouse_state           | string  | State of the warehouse                                                                                                                                                                                                    |
```
Comment (gueniai) [inline docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:919]: if we have these published in DB docs, let's link them

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md [hunk 7]
```diff
@@ -0,0 +1,1005 @@
...
+| tags                      | map     | A set of key-value pairs that will be tagged on all resources (e.g., AWS instances and EBS volumes) associated with this SQL warehouse.                                                                                   |                                                                                  
+| num_clusters              | long    | current number of clusters running for the service                                                                                                                                                                        |                                                                                                                                                                       
+| num_active_sessions       | long    | current number of active sessions for the warehouse                                                                                                                                                                       |
```
Comment (gueniai) [inline docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:924]: Might be useful to explain what a session is, or point to docs

**Review by gueniai — CHANGES_REQUESTED** [review]

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md (hunk 4 above)
Comment (aman-db) [reply docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:912]: done

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md (hunk 3 above)
Comment (aman-db) [reply docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:910]: done

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md (hunk 2 above)
Comment (aman-db) [reply docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:908]: done

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md (hunk 5 above)
Comment (aman-db) [reply docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:916]: I have fetched these values from API docs. Let me know if we can add the API docs link here

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md (hunk 6 above)
Comment (aman-db) [reply docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:919]: Again fetched these values from API docs

File: docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md (hunk 7 above)
Comment (aman-db) [reply docs-compose/overwatch_docs/content/DataEngineer/Definitions/071x.md:924]: fetched the description from API docs

**Review by gueniai — APPROVED** [review]

**Pull request merged**